### DISCUSSION TITLE
Correct way to add direct association

### DISCUSSION THREAD

**Discussion by webmogul1** [zwave-js/zwave-js#5320] [Q&A]
Is this the correct way to add direct association between a door sensor and a light switch?

It is working but not 100% reliably. Sometimes light does not turn on when I open the door.

![image](https://user-images.githubusercontent.com/5225701/211103648-bd104cfc-9661-4d0b-9c2c-3494717cccae.png)

0	On/Off	ZWAVE DONGLE (Server Room)	None  **(so homeassistant can be notified that light was turned on or off)**
0	On/Off	Pantry Light (Kitchen)	None	**(turns on or off the light)**

Not sure about target endpoints 

Nothing to add in the light switch
![image](https://user-images.githubusercontent.com/5225701/211104176-f1a251c7-0f2e-4d62-930b-b95c7d7cc601.png)

**Comment by robertsLando** [comment]
You add associations to device A in order to allow it to send actions to device B when an events happens on device A.

So if you want to control your light when the door opens you have to add an association on door sensor node with the light, if the light switch has multiple endpoints (multiple switchs in this case) you have to assign the correct target endpoint, same if the door has multiple ones. If it has only one endpoint it is correct to see None in target (default is root endpoint in that case)

**Comment by webmogul1** [comment]
For some reason, direct associarion is not working 100%.
Sometimes the door sensor does not turn on the switch.

Also, switch got stuck twice with a different association (two switches) and the only way to unstuck it was by killing the power to the switch.

**Reply by robertsLando** [reply]
This could be because there are communication issues in your network, what stick are you using? Do you have an extension cord?

**Comment by webmogul1** [comment]
Zooz zst10 Stick

Yes, I am using an extension cable for the stick.

But doesn't direct associations bypass the stick.

Anyway, I only having problems with direct association. Everything else works fine without glitches.

**Reply by robertsLando** [reply]
Ok so please attach zwave-js [driver logs](https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=generating-log-files)

**Reply by webmogul1** [reply]
Will do. I will post is next time there is an issue.

**Reply by webmogul1** [reply]
Please see attached. I chose to log just 2 nodes.
Node 19 is the Door Sensor
Node 10 is the switch
[zwavejs_2023-01-10.log](https://github.com/zwave-js/zwave-js-ui/files/10386198/zwavejs_2023-01-10.log)
